### PR TITLE
Vajira web services

### PR DESCRIPTION
**Pull request by vajirakarunathilake**

### CODE REVIEW

File: server/src/main/java/p2/dao/impl/UserDAO.java [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+
+public class UserDAO extends GenericDAO<User> implements IUserDAO {
+
+	public User findByEmailANDPassword(String email, String password) {
+		Session session = HibernateUtil.getSession();
+		User user = null;
 
-public class UserDAO extends GenericDAO<User> implements IUserDAO { 
+		try {
+			user =  (User) session.createQuery("FROM Interest WHERE email = " + email+" AND password =" + password);
```
Comment (mjrerle) [inline server/src/main/java/p2/dao/impl/UserDAO.java]: use query.setParameter here (might be open to SQL injections)

File: server/src/main/java/p2/dao/impl/UserDAO.java [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+
 import p2.dao.IUserDAO;
 import p2.model.User;
+import p2.util.HibernateUtil;
+
+public class UserDAO extends GenericDAO<User> implements IUserDAO {
+
+	public User findByEmailANDPassword(String email, String password) {
```
Comment (mjrerle) [inline server/src/main/java/p2/dao/impl/UserDAO.java]: can we call this method findByEmailAndPassword?

File: server/src/main/java/p2/service/impl/InterestService.java [hunk 3]
```diff
@@ -26,20 +26,31 @@ public static Interest findById(int id) {
...
 
+	public static int changeStatusNotEnoughInterest(int productId) {
+		return interestDAO.changeStatus(productId, InterestStatus.NOT_ENOUGH_INTEREST.value);
+	}
+
+	public static int changeStatusCancelledBySeller(int productId) {
+		return interestDAO.changeStatus(productId, InterestStatus.CANCELLED_BY_SELLER.value);
+	}
+
+	public static int changeStatusDealComplted(int productId) {
```
Comment (mjrerle) [inline server/src/main/java/p2/service/impl/InterestService.java]: change this to changeStatusToDealCompleted

File: server/src/main/java/p2/service/impl/InterestService.java [hunk 4]
```diff
@@ -26,20 +26,31 @@ public static Interest findById(int id) {
...
 			count += interest.getQuantity();
 		return count;
 
 	}
 
+	public static int changeStatusNotEnoughInterest(int productId) {
+		return interestDAO.changeStatus(productId, InterestStatus.NOT_ENOUGH_INTEREST.value);
+	}
+
+	public static int changeStatusCancelledBySeller(int productId) {
```
Comment (mjrerle) [inline server/src/main/java/p2/service/impl/InterestService.java]: change this to changeStatusToCancelledBySeller

File: server/src/main/java/p2/service/impl/InterestService.java [hunk 5]
```diff
@@ -26,20 +26,31 @@ public static Interest findById(int id) {
...
-	public static int getNumberOfInterest_ByProductId(int productId) {
+	public static int getNumberOfInterestByProductId(int productId) {
 		int count = 0;
 		for (Interest interest : findByProductId(productId))
 			count += interest.getQuantity();
 		return count;
 
 	}
 
+	public static int changeStatusNotEnoughInterest(int productId) {
```
Comment (mjrerle) [inline server/src/main/java/p2/service/impl/InterestService.java]: change this to changeStatusToNotEnoughInterest

File: server/src/main/java/p2/webservice/UserWebService.java [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+	public static void login(HttpServletRequest request, HttpServletResponse response) {
+
+		String email = request.getParameter("email");
+		String password = request.getParameter("password");
+
+		User user = UserService.findByEmailANDPassword(email, password);
+		try {
+			if (user != null) {
+				SessionVariableManager.addLoggedInUser(request, user);
+				logger.info("User Id: " + user.getId() + " Logged to the sysytem");
```
Comment (mjrerle) [inline server/src/main/java/p2/webservice/UserWebService.java]: change this to User <email> logged into the system

File: server/src/main/java/p2/webservice/UserWebService.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+				response.getWriter().append("Successfully Logged to the system").close();
```
Comment (mjrerle) [inline server/src/main/java/p2/webservice/UserWebService.java]: change this to Successfully logged in

File: server/src/main/java/p2/webservice/UserWebService.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+	public static void logout(HttpServletRequest request, HttpServletResponse response) {
+		try {
+			SessionVariableManager.removeLoggedInUser();
+			response.getWriter().close();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public static void insert(HttpServletRequest request, HttpServletResponse response) {
```
Comment (mjrerle) [inline server/src/main/java/p2/webservice/UserWebService.java:62]: we need simple validation of these parameters (all of them because of postman testing)

**Review by mjrerle — COMMENTED** [review]
don't worry about the validation: I will take care of it. fix the spelling mistakes and capitalization on the mentioned methods/variables and push it back to this branch.

File: server/src/main/java/p2/dao/impl/UserDAO.java (hunk 1 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/dao/impl/UserDAO.java]: Done

File: server/src/main/java/p2/dao/impl/UserDAO.java (hunk 2 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/dao/impl/UserDAO.java]: done

File: server/src/main/java/p2/service/impl/InterestService.java (hunk 3 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/service/impl/InterestService.java]: Done

File: server/src/main/java/p2/service/impl/InterestService.java (hunk 4 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/service/impl/InterestService.java]: Done

File: server/src/main/java/p2/service/impl/InterestService.java (hunk 5 above)
Comment (vajirakarunathilake) [reply server/src/main/java/p2/service/impl/InterestService.java]: Done

**Review by mjrerle — APPROVED** [review]

**Pull request merged**